torch_xla/csrc/ops/expand_dynamic.cpp

```diff
@@ -0,0 +1,81 @@
+#include "torch_xla/csrc/ops/expand_dynamic.h"
+
+#include "absl/strings/str_join.h"
+#include "tensorflow/compiler/xla/client/lib/constants.h"
+#include "tensorflow/compiler/xla/xla_client/debug_macros.h"
+#include "torch/csrc/lazy/core/helpers.h"
+#include "torch/csrc/lazy/core/util.h"
+#include "torch_xla/csrc/data_ops.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+
+namespace torch_xla {
+namespace {
+
+std::vector<xla::Shape> GetShapes(const torch::lazy::Value& input,
+                                  const std::vector<int64_t> upper_bounds,
+                                  const std::vector<bool> dynamic_dims) {
+  std::vector<xla::Shape> shapes;
+  shapes.push_back(GetXlaShape(input));
+  for (int i = 0; i < upper_bounds.size(); i++) {
+    shapes.push_back(
+        xla::ShapeUtil::MakeShape(GetXlaShape(input).element_type(),
+                                  {upper_bounds[i]}, {dynamic_dims[i]}));
+  }
+  return shapes;
+}
+
+xla::Shape NodeOutputShape(const torch::lazy::Value& input,
+                           const std::vector<int64_t> upper_bounds,
+                           const std::vector<bool> dynamic_dims) {
+  return xla::ShapeUtil::MakeShape(GetXlaShape(input).element_type(),
+                                   {upper_bounds}, {dynamic_dims});
+}
+
+std::vector<torch::lazy::Value> GetValues(
+    const torch::lazy::Value& input,
+    const std::vector<torch::lazy::Value> dimensions) {
```

Comment:
either pass `dimensions` as `const &` or pass it as value (drop the const). First one might be a better choice. If you choose the second one, no need to make a new vector.